src/Microsoft.OpenApi.OData.Reader/OpenApiConvertSettings.cs

```diff
@@ -188,6 +188,16 @@ public string PathPrefix
         /// </summary>
         public bool EnableDollarCountPath { get; set; } = true;
 
+        /// <summary>
+        /// Gets/sets a value indicating whether or not to include the OData type cast segments.
+        /// </summary>
+        public bool EnableODataTypeCast { get; set; } = true;
+
+        /// <summary>
+        /// Gets/sets a value indicating whether or not to require a derived types constraint to include the OData type cast segments.
+        /// </summary>
+        public bool RequireDerivedTypesConstraintForODataTypeCastSegments { get; set; } = true;
```

Comment:
It's better to disable it by default.

Reply:
this true by default means we won't add any path segment if no annotations are present